### PR TITLE
feat: add pycasbin adopter and test.

### PR DESCRIPTION
**Pull request by sheny1xuan**
### Description
+ I acciently delete this branch and close the [pr](https://github.com/casbin/casbin-cpp/pull/163).
+ [First commit](https://github.com/casbin/casbin-cpp/commit/5557955da80b86ac9c2212070e7a46192e1d3d6b) 
  + Fix `FileAdapter.SavePolicy()`
  + Add a new FileAdapter interface to use in python like [casbin::Config](https://github.com/casbin/casbin-cpp/blob/abaa2174f0a4c6df2a25c21c388c15939f472655/casbin/config/config.h#L55).
  + Fix `ArrayToString(const std::vector<std::string>& arr)`
    + `arr` usually be a line of policy keys without `p` or `g`, (e.g. `arr = {"alice", "data1", "read"}`).
    + Before `ArrayToString(const std::vector<std::string>& arr)`, `arr` will return `alice,  alice, data1, read`. `alice` will generate twice.
+ [Second commit](https://github.com/casbin/casbin-cpp/pull/164/commits/a0a1bcb84c3a8e7d66fbedf64f62bafc8921759a)
  + Add pycasbin file adopter and test.

### CODE REVIEW

**Comment by hsluoyz** [comment]
@EmperorYP7 please review

File: tests/python/test_adapter.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        self.ModelInit(basic_model_path)
+        self.Adapter.LoadPolicy(self.Model)
+
+        FilePolicies = [["alice", "data1", "read"],
+                        ["bob", "data2", "write"]]
+
+        self.assertTrue(self.Model.HasPolicy('p', 'p', FilePolicies[0]))
+        self.assertTrue(self.Model.HasPolicy('p', 'p', FilePolicies[1]))
+
+if __name__ == "__main__":
```
Comment (EmperorYP7) [inline tests/python/test_adapter.py:59]: I don't think we need this.

**Review by EmperorYP7 — CHANGES_REQUESTED** [review]

File: tests/python/test_adapter.py (hunk 1 above)
Comment (sheny1xuan) [reply tests/python/test_adapter.py:59]: Do you mean the `test_FileAdapterLoadPolicy` or `main block`? I think maybe I only need delete the `main block` and keep this test suite.

**Comment by sheny1xuan** [comment]
@EmperorYP7 please review.

File: tests/python/test_adapter.py (hunk 1 above)
Comment (EmperorYP7) [reply tests/python/test_adapter.py]: Yes, delete the `main` block

**Review by EmperorYP7 — APPROVED** [review]
LGTM!

**Pull request merged**